core/src/main/scala/org/apache/spark/sql/execution/ExistingPlans.scala

```diff
@@ -129,10 +131,9 @@ private[sql] object PartitionedPhysicalScan {
         ColumnTableScan(output, rdd, otherRDDs, numBuckets,
           partitionColumns, relation, allFilters, schemaAttributes)
       case r: SamplingRelation =>
-        if (r.isPartitioned) {
-          val columnScan = ColumnTableScan(output, rdd, Nil, numBuckets,
+       if (r.isReservoirAsRegion) {
+          SampleColumnTableScan(output, rdd, Nil, numBuckets,
```

Comment:
As for my comment in other PR, I am confused what rdds are being passed in SampleColumnTableScan. Are we ignoring any rdd?

SampleColumnTableScan should be equipped to handle all three RDDs, but if you plan to take this afterwards, please comment out code that put rows in rowbuffer and also create a defect for same.